plugins/minions.js

```diff
@@ -0,0 +1,30 @@
+var request = require('request');
+
+module.exports = {listeners: [
+{
+    type: "equals",
+    query: ".minion",
+    callback: function(reply, message){
+        var url = Math.random()<.5 ? "https://www.reddit.com/r/MinionHate/top/.json?sort=top&t=week" : "https://www.reddit.com/r/WackyTicTacs/top/.json?sort=top&t=week";
+        request(url, function(err, res, body){
+            try{
+                var posts = JSON.parse(body).data.children;
+                var index = Math.floor(Math.random() * posts.length);
+                while(!posts[index].data.url){
```

Comment:
I'm not intimately familiar with the Reddit API and its response format, but this looks like a prime candidate for an endless loop, depending on the data received. What happens if there are no URLs to be found?